feat(core): Add config option to skip files considered for imports

sauntimo opened the pull request:
### Background
- closes #136 
- When functions are imported into tests, (and depending on the local tsconfig settings), ts-prune may no longer consider them dead code, despite them not being use "in anger" anywhere in the project. In these instances it would be useful to be able to specify a pattern (eg `.test.ts`) whereby matching source files would be ignored when determining if code is dead.

### Changes
- adds `-s --skip [regexp]` flag which allows users to specify a pattern for source files to ignore
- adds two tests for `getPotentiallyUnused` to ensure that it's working with and without a skip pattern
- adds a skip pattern case to the integration test suite
- fixes delete command and gitignore for the outfile generated by the integration test, and deletes it from git history.

nadeesha commented on src/analyzer.ts:
```diff
@@ -191,26 +192,33 @@ const getDefinitelyUsed = (file: SourceFile): IAnalysedResult[] => ([
-  const referenced = file
-    .getReferencingNodesInOtherSourceFiles()
-    .reduce(
+  let references = file.getReferencingNodesInOtherSourceFiles();
```
Could you kindly refactor this to be a function, so we don't have conditional logic within this function, and eliminate the `let`?

Something like:

```ts
const getReferences = (originalList, regex) => {
  if (skipPattern) {
    return ...
  } else {
    return originalList;
  }
}
```

So that this section becomes:

```ts
const referenced = getReferences(file.getReferencingNodesInOtherSourceFiles(), skipPattern).reduce(...)
```

sauntimo replied:
Haha, I actually considered this! Yeah, absolutely.

nadeesha reviewed (commented):
Very nice! Almost there. 

Could you add a test case to the integration tests as well?

nadeesha commented on README.md:
```diff
@@ -45,6 +45,7 @@ ts-prune supports CLI and file configuration via [cosmiconfig](https://github.co
 - `-p, --project` - __tsconfig.json__ path(`tsconfig.json` by default)
 - `-i, --ignore` - errors ignore RegExp pattern
 - `-e, --error` - return error code if unused exports are found
+- `-s, --skip` - skip these files when determining whether code is used - Regexp Pattern
```
```suggestion
- `-s, --skip` - regex pattern to skip these files when determining whether code is used. (For example, `*.test.ts?` will stop ts-prune from considering and export in test file usages)
```

sauntimo replied:
Done (except "an" instead of "and"!)

nadeesha commented on src/configurator.ts:
```diff
@@ -23,6 +25,7 @@ export const getConfig = () => {
     .option('-p, --project [project]', 'TS project configuration file (tsconfig.json)', 'tsconfig.json')
     .option('-i, --ignore [regexp]', 'Path ignore RegExp pattern')
     .option('-e, --error', 'Return error code if unused exports are found')
+    .option('-s, --skip [regexp]', 'skip files when determining code use - Regexp pattern')
```
```suggestion
    .option('-s, --skip [regexp]', 'skip these files when determining whether code is used')
```

sauntimo commented:
> Could you add a test case to the integration tests as well?

Ah yeah, good call. I'll have a look at this in the week. Thanks for your comments, they all make sense to me 👍

sauntimo commented:
@nadeesha hey, sorry I appreciate you're probably busy but I'd love to get this finished off if you get a moment 😁 No worries if you don't get a chance to look at it though. Have a nice weekend!

nadeesha reviewed (commented):
@sauntimo Excellent. I just took a look and almost everything looks great.

I saw however that the integration tests are failing. Could you kindly look into that, and we can merge it right after?

sauntimo commented:
Oh, that's embarrassing! Thanks for pointing that out, I'll see if can fix this over the weekend.

sauntimo commented:
@nadeesha I think I've fixed the integration test now. I noticed that the generated `outfile` had somehow gotten git tracked so I updated the command to remove it, and the gitignore which should have stopped it from being tracked. I realised that I had added my case to _that_ `outfile` rather than the `outfile.base` 🤦. Hopefully everything is working now!

nadeesha commented:
@sauntimo Thanks for addressing the feedback. This is now merged.

sauntimo commented:
@nadeesha awesome - thanks for your patience & help with it! ☺️
